src/box/xrow.c

```diff
@@ -1714,7 +1714,7 @@ xrow_decode_error(const struct xrow_header *row)
 		} else if (key == IPROTO_ERROR) {
 			struct error *e = error_unpack_unsafe(&pos);
 			if (e == NULL)
-				goto error;
+				return;
```

Comment:
For other errors encountered while parsing a packet, we set the diag to the code from the packet header and an empty message. Why do it differently in this case?

Reply:
Removed.